rust/lance-core/src/io/writer/statistics.rs

```diff
@@ -97,57 +97,82 @@ where
     T::Native: Bounded,
     datafusion_common::scalar::ScalarValue: From<<T as ArrowPrimitiveType>::Native>,
 {
-    let (min_value, max_value, null_count) = get_primitive_statistics::<T>(arrays);
+    debug_assert!(!arrays.is_empty());
+    // Note: we take data_type off of arrays instead of using T::DATA_TYPE because
+    // there might be parameters like timezone for temporal types that need to
+    // be propagated.
+    let (min_value, max_value, null_count) = compute_primitive_statistics::<T>(arrays);
     StatisticsRow {
         null_count,
-        min_value: ScalarValue::from(min_value),
-        max_value: ScalarValue::from(max_value),
+        min_value: ScalarValue::new_primitive::<T>(Some(min_value), arrays[0].data_type()),
+        max_value: ScalarValue::new_primitive::<T>(Some(max_value), arrays[0].data_type()),
     }
 }
 
+fn compute_float_statistics<T: ArrowNumericType>(
+    arrays: &[&ArrayRef],
+) -> (T::Native, T::Native, i64)
+where
+    T::Native: Float,
+{
+    let mut min_value = T::Native::infinity();
+    let mut max_value = T::Native::neg_infinity();
```

Comment:
It turns out that `T::MIN` and `T::MAX` given the smallest and largest *finite* value, whereas we want `-Inf` and `Inf`. So had to specialize this method.